bpf/init.sh

```diff
@@ -283,11 +283,24 @@ function bpf_load_cgroups()
 	rm -f "$BPFFS_ROOT/cilium/socketlb/links/cgroup/$NAME"
 	set -e
 
-	if ! bpftool cgroup attach "$CGRP" "$WHERE" pinned "$TMP_FILE"; then
+	if bpftool cgroup attach "$CGRP" "$WHERE" pinned "$TMP_FILE"; then
 		rm -f "$TMP_FILE"
-		cilium bpf migrate-maps -e "$OUT" -r 1
-		return 1
+		return 0
+	fi
+
+	# Program might've been attached in multi-mode by a newer version of Cilium or
+	# by another tool. This means 'bpftool cgroup attach' won't succeed unless
+	# any/all attached programs are removed.
+	bpf_clear_cgroups "$CGRP" "$WHERE" "$NAME"
+
+	if bpftool cgroup attach "$CGRP" "$WHERE" pinned "$TMP_FILE"; then
+		rm -f "$TMP_FILE"
```

Comment:
I note you're removing the `$TMP_FILE` even on success, here and above. I don't think this is an issue, but just raising it as an observation.

Reply:
> Not sure if bpftool cgroup attach already does this behind the scenes.

I don't think so